### PR TITLE
Add API

### PR DESCRIPTION
**Pull request by rediska0123**

### CODE REVIEW

File: api/api.go [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+	w.WriteHeader(http.StatusOK)
+}
+
+type getLinksRequestModel struct {
+	token string
+}
+
+func (a *Api) getLinks(w http.ResponseWriter, r *http.Request) {
+	var m getLinksRequestModel
+	if err := json.NewDecoder(r.Body).Decode(&m); err != nil {
```
Comment (M0nsteR) [inline api/api.go:92]: Please note that this method uses a GET verb. Requiring a request body here is a bit of a gray area.

File: api/api.go [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+	var acc *usecases.Account = nil
+	if m.token != nil {
+		a, err := tokenToAccount(*m.token)
+		acc = &a
+		if err != nil {
+			w.WriteHeader(http.StatusBadRequest)
+			return
+		}
+	}
+	id, err := a.UseCases.CreateSnippet(acc, usecases.CodeSnippet{Code: m.code, Lang: m.lang, Lifetime: m.lifetime})
```
Comment (M0nsteR) [inline api/api.go]: Are you sure you want to pass a CodeSnippet struct to the Create func before actually making all necessary checks of input values? Please make it consistent with the rest of your apis, eg. when creating an account you pass the raw arguments (login + password strings), not a struct, to the function.

File: main.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+		Addr:         "localhost:8080",
+		ReadTimeout:  10 * time.Second,
+		WriteTimeout: 10 * time.Second,
+
+		Handler: service.Router(),
+	}
+	err := server.ListenAndServe()
+	if err != nil {
+		panic(err)
+	}
```
Comment (M0nsteR) [inline main.go:28]: It might be good to display some initial greeting from your running server here.

File: usecases/usecases.go [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+func (User) GetMyLinks(a Account) ([]string, error) {
+	// TODO
+	fmt.Printf("GetMyLinks: %s", a.Id)
+	return []string{"a", "b", "c"}, nil
+}
+
+func (User) CreateSnippet(a *Account, cs CodeSnippet) (string, error) {
+	// TODO
+	fmt.Printf("CrateLink: %s %s", a.Id, cs.Code)
```
Comment (M0nsteR) [inline usecases/usecases.go]: a type: crate -> create

**Review by M0nsteR — CHANGES_REQUESTED** [review]

File: api/api.go (hunk 1 above)
Comment (rediska0123) [reply api/api.go:92]: Made it POST

**Review by M0nsteR — APPROVED** [review]
Looks good as a first draft, thank you.

**Pull request merged**